HiP-UserStore/Migrations/Migration1PropertyChanged.cs

```diff
@@ -0,0 +1,131 @@
+                            };
+                            propEvents = EntityManager.CompareEntities(currentArgs, newUserArgs, ev.GetEntityType(), ev.Id, ev.UserId);
+                            argumentDictionary[(ev.GetEntityType(), ev.Id)] = newUserArgs;
+                            break;
+
+                        case UserPhotoUploaded ev:
+                            timestamp = ev.Timestamp;
+                            var currentPhotoArgs = argumentDictionary[(ev.GetEntityType(), ev.Id)] as UserArgs2;
+                            if (currentPhotoArgs == null)
+                                currentPhotoArgs = new UserArgs2();
+                            newUserArgs = new UserArgs2
+                            {
+                                ProfilePicturePath = ev.Path,
+                                FirstName = currentPhotoArgs.FirstName,
+                                LastName = currentPhotoArgs.LastName,
+                                Email = currentPhotoArgs.Email,
+                                Password = currentPhotoArgs.Password,
+                                StudentDetails = currentPhotoArgs.StudentDetails,
+                                UserId = currentPhotoArgs.UserId
+                            };
+                            propEvents = EntityManager.CompareEntities(currentPhotoArgs, newUserArgs, ev.GetEntityType(), ev.Id, ev.UserId);
+                            argumentDictionary[(ev.GetEntityType(), ev.Id)] = newUserArgs;
+                            break;
+
+                        case UserStudentDetailsUpdated ev:
+                            timestamp = ev.Timestamp;
+                            var currentUserArgs = argumentDictionary[(ev.GetEntityType(), ev.Id)] as UserArgs2;
+                            if (currentUserArgs == null)
+                                currentUserArgs = new UserArgs2();
+                            newUserArgs = new UserArgs2
+                            {
+                                StudentDetails = new StudentDetails(new StudentDetailsArgs
+                                {
+                                    CurrentDegree = ev.Properties.CurrentDegree,
+                                    CurrentSemester = ev.Properties.CurrentSemester,
+                                    Discipline = ev.Properties.Discipline
+                                }),
+                                FirstName = currentUserArgs.FirstName,
+                                LastName = currentUserArgs.LastName,
+                                Email = currentUserArgs.Email,
+                                Password = currentUserArgs.Password,
+                                ProfilePicturePath = currentUserArgs.ProfilePicturePath,
+                                UserId = currentUserArgs.UserId
+                            };
+                            propEvents = EntityManager.CompareEntities(currentUserArgs, newUserArgs, 
```

Comment:
You should compare the two StudentDetails objects here instead of the UserArgs2 objects. This would then lead to a PropertyChangedEvent with ResourceType StudentDetails. You would also need to create CreatedEvent for a StudentDetails entity , if this is the first updated event.